### PR TITLE
gcovr documentation improvements

### PR DESCRIPTION
**Pull request by tyrantlizard77**
- Add gcovr to our set of packages installed with pip
- Fix numerous issues in the documentation for using gcovr to produce a report for a single application
- Expand documentation for producing coverage reports with sanitycheck

### CODE REVIEW

**Review by nashif — APPROVED** [review]

File: doc/guides/coverage.rst [hunk 1]
```diff
@@ -44,25 +44,39 @@ To report the coverage for the particular test application set :option:`CONFIG_C
...
+1. Build the code with CONFIG_COVERAGE=y. Some boards like qemu_x86_coverage
+   automatically enable this, but for boards that do not you will need to
+   enable the configuration manually::
 
      $ cmake -DBOARD=mps2_an385 -DCONFIG_COVERAGE=y ..
+     $ make
 
-#. Store the build and run output on to a log file::
+#. Capture the emulator output into a log file. You may need to terminate
+   the emulator with ``Ctrl-A X`` for this to complete after the coverage dump
```
Comment (dbkinder) [inline doc/guides/coverage.rst]: ... or you can use `` :kbd:`Ctrl-A X` ``

**Review by dbkinder — APPROVED** [review]
One optional tweak, otherwise LGTM

File: doc/guides/coverage.rst [hunk 2]
```diff
@@ -44,25 +44,39 @@ To report the coverage for the particular test application set :option:`CONFIG_C
...
-     $ find -iregex ".*gcov"
+#. Create an output directory for the reports::
 
-#. Run gcovr to get the reports::
+     $ mkdir -p gcov_report
 
-     $ gcovr -r . --html -o gcov_report/coverage.html --html-details --gcov-executable <gcov_path_in_SDK>
+#. Run ``gcovr`` to get the reports::
+
+     $ gcovr -r $ZEPHYR_BASE zephyr --html -o gcov_report/coverage.html --html-details --gcov-executable <gcov_path_in_SDK>
```
Comment (nashif) [inline doc/guides/coverage.rst]: remove that extra zephyr

**Review by nashif — CHANGES_REQUESTED** [review]

File: doc/guides/coverage.rst (hunk 2 above)
Comment (andrewboie) [reply doc/guides/coverage.rst]: @nashif no.

That zephyr tells gcovr where to look for gcda/gcno files. It should not be searching for them in $ZEPHYR_BASE.

Comment (andrewboie) [reply doc/guides/coverage.rst]: usage: gcovr [options] [search_paths...]

Comment (nashif) [reply doc/guides/coverage.rst]:
```
Traceback (most recent call last):
  File "/usr/bin/gcovr", line 11, in <module>
    load_entry_point('gcovr==4.1', 'console_scripts', 'gcovr')()
  File "/usr/lib/python3.7/site-packages/gcovr/__main__.py", line 548, in main
    datafiles = get_datafiles(options.search_paths, options)
  File "/usr/lib/python3.7/site-packages/gcovr/gcov.py", line 53, in get_datafiles
    ".*\.gc(da|no)$", dir_, exclude_dirs=options.exclude_dirs)
  File "/usr/lib/python3.7/site-packages/gcovr/utils.py", line 153, in search_file
    raise IOError("Unknown directory '" + path + "'")
OSError: Unknown directory 'zephyr'
```

Comment (nashif) [reply doc/guides/coverage.rst]: i noticed you changed it to "." now...

Comment (andrewboie) [reply doc/guides/coverage.rst]: @nashif these instructions are from the out directory, of course zephyr/ exists

**Review by nashif — APPROVED** [review]

**Pull request merged**